### PR TITLE
Release New-Relic CLI for stable version 2024-01-01

### PR DESCRIPTION
**Pull request by dipeshbhakat-microsoft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yonzhan** [comment]
Please fix CI issues

File: src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/_create.py [hunk 1]
```diff
@@ -51,6 +51,8 @@ def _build_arguments_schema(cls, *args, **kwargs):
             required=True,
         )
         _args_schema.resource_group = AAZResourceGroupNameArg(
+            options=["--resource-group"],
```
Comment (jsntcy) [inline src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/_create.py:54]: It seems this change introduces a breaking change (Previously it supported both "--resource-group" and "--g", but now it only supports "--resource-group"). Please check other breaking changes as well.

**Comment by jsntcy** [comment]
> ❌Azure CLI Extensions Breaking Change Test

@dipeshbhakat-microsoft,
It seems there are lots of breaking changes introduced by this PR and I think many are unexpected, for example previously supported both "--resource-group" and "--g", but now only support "--resource-group".
If you do want those breaking changes, you need release them in breaking change window.

@necusjz, @AllyW, please help confirm.

File: src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/_create.py (hunk 1 above)
Comment (dipeshbhakat-microsoft) [reply src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/_create.py:54]: This is not expected. I did it by mistake will revert this change.

Comment (dipeshbhakat-microsoft) [reply src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/_create.py]: @jsntcy I have reverted the changes which were breaking changes. Please have a look if anything else is breaking.

Comment (necusjz) [reply src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/_create.py]: @dipeshbhakat-microsoft still two?

image

Comment (dipeshbhakat-microsoft) [reply src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/_create.py]: I have now added the remaining test cases and fixed the breaking changes

Comment (dipeshbhakat-microsoft) [reply src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/_create.py]: @necusjz all work from my side is done now. Let me know if anything else is required from my side.

**Comment by dipeshbhakat-microsoft** [comment]
> ## ⚠️ Suggestions
> ### Module: new-relic
> * Update version to `1.0.0b2` in setup.py
> * Set `azext.isPreview` to `true` in azext_new-relic/azext_metadata.json if not exists
> 
> #### Notes
> * Stable/preview tag is inherited from last release. If needed, please add `stable`/`preview` label to modify it.
> * Major/minor/patch/pre increment of version number is calculated by pull request code changes automatically. If needed, please add `major`/`minor`/`patch`/`pre` label to adjust it.
> * For more info about extension versioning, please refer to [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

This is a stable version release so updated the version to 1.0.0 and removed `azext.isPreview` = `true` from azext_new-relic/azext_metadata.json.

Please let me know if anything else is required

**Comment by dipeshbhakat-microsoft** [comment]
@jsntcy @necusjz can we please approve and merge it if anything is not pending.

**Comment by dipeshbhakat-microsoft** [comment]
@kairu-ms / @jsntcy can we please remove **[do-not-merge](https://github.com/Azure/azure-cli-extensions/labels/do-not-merge)** lable and complete the review. Its pending from long.

**Review by kairu-ms — APPROVED** [review]
LGTM

**Pull request merged**